spec/httpserver/routing/router_spec.clj

```diff
@@ -0,0 +1,30 @@
+(ns httpserver.routing.router-spec
+  (:require [speclj.core :refer :all]
+            [httpserver.routing.router :as router]
+            [httpserver.handlers.handler]))
+
+(import '(com.td.HttpServer HttpServer))
+
+(describe "Checking if this is the route it should be looking for"
+  (it "returns false if the method of the current route being checked does not match the method it should be looking for"
+    (should= false (router/check-route "PUT" "/path" {:method "GET" :path "/path"})))
+  (it "returns false if the path of the current route being checked does not match the path it should be looking for"
+    (should= false (router/check-route "GET" "/" {:method "GET" :path "/path"})))
+  (it "returns false if neither the method nor path do not match the ones it should be looking for"
+    (should= false (router/check-route "PUT" "/" {:method "GET" :path "/path"})))
+  (it "returns true if the method and path match the ones it should be looking for"
+    (should= true (router/check-route "GET" "/path" {:method "GET" :path "/path"}))))
```

Comment:
This test is good evidence of TDD-ing this feature, but I think `router/check-route` should ultimately be private. This is a situation where you would write out the specs as you test-drive the feature, then write tests for the "higher up" function (`router/find-route`), then go back and make this function private and delete these tests.

In fact, `router/find-route` might warrant being private too—it only gets called by `router/get-handler`, correct? So maybe you should only be testing this namespace via `get-handler`...

PS: to make a function private, define it with a hyphen: `(defn- this-is-private ...)`